src/components/theme/__tests__/ThemeProvider.test.tsx

```diff
@@ -0,0 +1,65 @@
+import {render, screen} from '@testing-library/react';
+import React from 'react';
+import {mockMatchMedia} from '../../../../test-utils/mocks/mockMatchMedia';
+import {ThemeProvider} from '../ThemeProvider';
+import {useTheme} from '../useTheme';
+import {useThemeValue} from '../useThemeValue';
+
+function ThemeDisplay() {
+    const [theme] = useTheme();
+    const themeValue = useThemeValue();
+
+    return (
+        <>
+            Theme: {theme}, theme value: {themeValue}
+        </>
+    );
+}
+
+test('should set "system" theme by default', () => {
+    const mock = mockMatchMedia();
+
+    render(
+        <ThemeProvider>
+            <ThemeDisplay />
+        </ThemeProvider>,
+    );
+
+    expect(screen.getByText('Theme: system, theme value: light')).toBeInTheDocument();
+
+    mock.uninstall();
+});
+
+test('should change theme value on system theme change', async () => {
+    const mock = mockMatchMedia();
+
+    render(
+        <ThemeProvider>
+            <ThemeDisplay />
+        </ThemeProvider>,
+    );
+
+    mock.changeMedia(true);
+
+    const el = await screen.findByText('Theme: system, theme value: dark');
+    expect(el).toBeInTheDocument();
+
+    mock.uninstall();
+});
+
+test('should keep selected theme when system theme changes', async () => {
+    const mock = mockMatchMedia();
+
+    render(
+        <ThemeProvider theme="light">
+            <ThemeDisplay />
+        </ThemeProvider>,
+    );
+
+    mock.changeMedia(true);
+
+    const el = await screen.findByText('Theme: light, theme value: light');
+    expect(el).toBeInTheDocument();
```

Comment:
To say it doesn't change, we also need to check `screen.getByText('Theme: light, theme value: light');` before `mock.changeMedia`, right?

Reply:
We already assert it in previous test

Reply:
We assert it for `system` them

Reply:
Fixed